### PR TITLE
Use nodePortOp for allocating healthCheck nodePort

### LINKED ISSUE #49999: Allocate healthcheck nodeport the same way as common nodeports
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
healthcheck nodeport allocation doesn't have roll-back, this may lead to nodeport leak.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
/sig network
/assign

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**: Allocate healthCheck nodePort via nodePortOp so that we won't leak port on failure.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #49999

**Special notes for your reviewer**:
/assign @xiangpengzhao @thockin 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/registry/core/service/rest.go [hunk 1]
```diff
@@ -257,19 +257,15 @@ func (rs *REST) healthCheckNodePortUpdate(oldService, service *api.Service) (boo
...
 	// Free the existing healthCheckNodePort and clear the HealthCheckNodePort field.
 	case neededHealthCheckNodePort && !needsHealthCheckNodePort:
 		glog.Infof("Transition to non LoadBalancer type service or LoadBalancer type service with ExternalTrafficPolicy=Global")
-		err := rs.serviceNodePorts.Release(int(oldHealthCheckNodePort))
-		if err != nil {
-			glog.Warningf("error releasing service health check %s node port %d: %v", service.Name, oldHealthCheckNodePort, err)
-			return false, errors.NewInternalError(fmt.Errorf("failed to free health check nodePort: %v", err))
-		}
+		nodePortOp.ReleaseDeferred(int(oldHealthCheckNodePort))
 		glog.Infof("Freed health check nodePort: %d", oldHealthCheckNodePort)
```
Comment (xiangpengzhao) [inline pkg/registry/core/service/rest.go]: Do we still want to keep this log? We are not sure if the `oldHealthCheckNodePort` will be actually freed at this point.

**Review by xiangpengzhao — COMMENTED** [review]
Just one nit, otherwise LGTM. Thanks for the fix :)

File: pkg/registry/core/service/rest.go (hunk 1 above)
Comment (MrHohn) [inline pkg/registry/core/service/rest.go]: Yeah that sentence may not be accurate anymore, and V(0) seems too verbose. Revised a bit and increased the verbose level.

**Comment by xiangpengzhao** [comment]
/lgtm
Though I believe bot won't allow me to label this :)

**Comment by cblecker** [comment]
/lgtm
@MrHohn Do you think this is work picking back to 1.7?

**Comment by MrHohn** [comment]
> Do you think this is work picking back to 1.7?

Yeah seems to be a small enough fix that could be cherrypicked.

**Comment by yastij** [comment]
@MrHohn needs rebase. also ping @wojtek-t for cherrypicking to 1.7

**Comment by MrHohn** [comment]
Sorry for the delay, rebased.

**Comment by MrHohn** [comment]
Failures don't have build logs.

/retest

**Pull request merged**